features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/ApiKeyRestriction.jsx

```diff
@@ -0,0 +1,335 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState } from "react";
+import { injectIntl } from "react-intl";
+import PropTypes from "prop-types";
+import { withStyles } from "@material-ui/core/styles";
+import TextField from "@material-ui/core/TextField";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import IconButton from "@material-ui/core/IconButton";
+import DeleteIcon from "@material-ui/icons/Delete";
+import Tooltip from "@material-ui/core/Tooltip";
+import Grid from "@material-ui/core/Grid";
+import Fab from "@material-ui/core/Fab";
+import AddIcon from "@material-ui/icons/Add";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
+import ListItemText from "@material-ui/core/ListItemText";
+import Box from "@material-ui/core/Box";
+import Radio from "@material-ui/core/Radio";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import FormControl from "@material-ui/core/FormControl";
+import Button from "@material-ui/core/Button";
+
+// Styles for Grid and Paper elements
+const styles = (theme) => ({
+  FormControl: {
+    "margin-bottom": "8px",
+    width: "100%",
+    padding: "0px 10px",
+  },
+});
+
+/**
+ * Used to display IP restrictions in generate api key UI
+ */
+const apiKeyRestrictions = (props) => {
+  const [invalidIP, setInvalidIP] = useState(false);
+  const [invalidReferer, setInvalidReferer] = useState(false);
+
+  const {
+    classes,
+    newIP,
+    updateNewIp,
+    ipList,
+    updateIpList,
+    restrictSchema,
+    updateRestrictSchema,
+    newReferer,
+    updateNewReferer,
+    refererList,
+    updateRefererList,
+  } = props;
+
+  const onRefererTextUpdate = (e) => {
+    updateNewReferer(e.target.value.trim());
+    if (e.target.value.trim() == "") {
+      setInvalidReferer(false);
+    }
+  };
+
+  const addRefererItem = () => {
+    if (newReferer != null && newReferer != "") {
+      setInvalidReferer(false);
+      refererList.push(newReferer.trim());
+      updateRefererList(refererList);
+      updateNewReferer("");
+    } else {
+      setInvalidReferer(true);
+    }
+  };
+
+  const deleteRefererItem = (refererItem) => {
+    refererList.splice(refererList.indexOf(refererItem), 1);
+    updateRefererList(refererList);
+  };
+
+  const onIpTextUpdate = (e) => {
+    updateNewIp(e.target.value.trim());
+    if (e.target.value.trim() == "") {
+      setInvalidIP(false);
+    }
+  };
+
+  const addIpItem = () => {
+    if (newIP != null && newIP != "") {
```

Comment:
Shal we use `!==` here
example.
```javascript
12 != '12' // false
```

This evaluates to `false`

Reply:
fixed